src/pages/index.js

```diff
@@ -2,51 +2,137 @@ import "./index.css";
 import Section from "../components/Section.js";
-import initialCards from "../utils/constants.js";
 import PopupWithImage from "../components/PopupWithImage.js";
 import PopupWithForm from "../components/PopupWithForm.js";
 import UserInfo from "../components/UserInfo.js";
+import Api from "../components/Api.js";
+import PopupDeleteCard from "../components/PopupDeleteCard.js";
 
 //Buttons
 const profileEditButton = document.querySelector("#profile-edit-btn");
 const addButton = document.querySelector("#add-button");
+const editAvatarButton = document.querySelector(".avatar-edit-button");
 
-function addCard (cardDataObj) {
-  const cardInstance = new Card(cardDataObj, "#card-template", (event) => {
-    imageModalHandler.open(event);
-  });
-  
-  const domCard = cardInstance.returnCard();
-
-  gridHandler.addItem(domCard);
-}
+//modalHandlers
+const deleteModalHandler = new PopupDeleteCard("#delete-card-modal");
+deleteModalHandler.setEventListeners();
 
-const profileInfoSelectors = {
-  nameSelector: ".profile__user-name",
-  jobSelector: ".profile__job"
-}
-
-const profileInfo = new UserInfo(profileInfoSelectors);
-
-//Modal Handlers
 const imageModalHandler = new PopupWithImage("#image-modal");
 imageModalHandler.setEventListeners();
 
 const profileModalHandler = new PopupWithForm("#profile-edit-modal", (data) => {
-  profileInfo.setUserInfo(data)
+  profileModalHandler.changeButtonText(true);
+  api.updateProfileInfo(data)
+  .then(json => {
+    profileInfo.setUserInfo(json);
+  })
+  .catch(err => {
+    console.error(err);
+  })
+  .finally(() => {
+    profileModalHandler.changeButtonText(false);
+  })
 });
 profileModalHandler.setEventListeners();
 
-const cardModalHandler = new PopupWithForm("#add-card-modal", addCard);
+const avatarModalHandler = new PopupWithForm("#avatar-edit-modal", (data) => {
+  avatarModalHandler.changeButtonText(true);
+  api.updateAvatar(data)
+  .then(json => {
+    api.updateAvatar(json);
+    profileInfo.setAvatar(json);
+  })
+  .catch(err => {
+    console.error(err);
+  })
+  .finally(() => {
+    avatarModalHandler.changeButtonText(false);
+  })
+});
+avatarModalHandler.setEventListeners();
+
+const cardModalHandler = new PopupWithForm("#add-card-modal",
+  data => {
+    cardModalHandler.changeButtonText(true);
+    api.addCard(data)
+    .then(json => {
+      addCard(json);
+    })
+    .catch(err => {
+      console.error(err);
+    })
+    .finally(() => {
+      cardModalHandler.changeButtonText(false);
+    })
+  }
+);
 cardModalHandler.setEventListeners();
 
-//Render Initial cards onto the page
-const gridHandler = new Section({
-  items: initialCards,
+//api for handling http requests
+const api = new Api();
+
+const section = new Section({
+  items: [],
   renderer: addCard
-}, ".cards__grid");
+}, ".cards__grid")
 
-gridHandler.renderItems();
+//used to render initial cards
+function addCard (data) {
+  const card = new Card(data, "#card-template",
```

Comment:
It's highly recommended to make a separate function `createCard` for creating a card not to copy-paste the code

```js		

function createCard(item) {
    // here you create a card
  return cardElement
}

```

You need to separate the logics of creating a card and adding it to the DOM because you can create a card and add with `append` or `prepend` or somehow else.



```js

function addCard (data) {
  const card = createCard(data)
  section.addItem(card);
}


```


`.returnCard()` will be inside it